web/pgadmin/dashboard/static/js/SystemStats/CPU.jsx

```diff
@@ -0,0 +1,301 @@
+  const [processCpuUsageStats, setProcessCpuUsageStats] = useState([]);
+
+  const [, setCounterData] = useState({});
+
+  const [pollDelay, setPollDelay] = useState(5000);
+
+  const [errorMsg, setErrorMsg] = useState(null);
+  const [chartDrawnOnce, setChartDrawnOnce] = useState(false);
+
+  const tableHeader = [
+    {
+      Header: 'PID',
+      accessor: 'pid',
+      sortable: true,
+      resizable: true,
+      disableGlobalFilter: false,
+    },
+    {
+      Header: 'Name',
+      accessor: 'name',
+      sortable: true,
+      resizable: true,
+      disableGlobalFilter: false,
+    },
+    {
+      Header: 'CPU Usage',
+      accessor: 'cpu_usage',
+      sortable: true,
+      resizable: true,
+      disableGlobalFilter: false,
+    },
+  ];
+
+  useEffect(()=>{
+    let calcPollDelay = false;
+    if(prevPrefernces) {
+      if(prevPrefernces['cu_stats_refresh'] != preferences['cu_stats_refresh']) {
+        cpuUsageInfoReduce({reset: chartsDefault['cu_stats']});
+        calcPollDelay = true;
+      }
+      if(prevPrefernces['la_stats_refresh'] != preferences['la_stats_refresh']) {
+        loadAvgInfoReduce({reset: chartsDefault['la_stats']});
+        calcPollDelay = true;
+      }
+      if(prevPrefernces['pcu_stats_refresh'] != preferences['pcu_stats_refresh']) {
+        setProcessCpuUsageStats({reset: chartsDefault['pcu_stats']});
+        calcPollDelay = true;
+      }
+    } else {
+      calcPollDelay = true;
+    }
+    if(calcPollDelay) {
+      const keys = Object.keys(chartsDefault);
+      const length = keys.length;
+      if(length == 1){
+        setPollDelay(
+          preferences[keys[0]+'_refresh']*1000
+        );
+      } else {
+        setPollDelay(
+          getGCD(Object.keys(chartsDefault).map((name)=>preferences[name+'_refresh']))*1000
+        );
+      }
+    }
+  }, [preferences]);
+
+  useEffect(()=>{
+    /* Charts rendered are not visible when, the dashboard is hidden but later visible */
+    if(pageVisible && !chartDrawnOnce) {
+      setChartDrawnOnce(true);
+    }
+  }, [pageVisible]);
+
+  useInterval(()=>{
+    const currEpoch = getEpoch();
+    if(refreshOn.current === null) {
+      let tmpRef = {};
+      Object.keys(chartsDefault).forEach((name)=>{
+        tmpRef[name] = currEpoch;
+      });
+      refreshOn.current = tmpRef;
+    }
+
+    let getFor = [];
+    Object.keys(chartsDefault).forEach((name)=>{
+      if(currEpoch >= refreshOn.current[name]) {
+        getFor.push(name);
+        refreshOn.current[name] = currEpoch + preferences[name+'_refresh'];
+      }
+    });
+
+    let path = getStatsUrl(sid, did, getFor);
+    if (!pageVisible){
+      return;
+    }
+    axios.get(path)
+      .then((resp)=>{
+        let data = resp.data;
+        setErrorMsg(null);
+        if(data.hasOwnProperty('cu_stats')){
+          let new_cu_stats = {
+            'User Normal': data['cu_stats']['usermode_normal_process_percent']?data['cu_stats']['usermode_normal_process_percent']:0,
```

Comment:
probably use gettext here and for others.

Reply:
@adityatoshniwal gettext for keys? eg User Normal, User Niced?

Reply:
If they're used to display text on the UI then yes.

Reply:
No actually they are numeric values. We are passing them in the StreamingChart.

Reply:
So, basically I should wrap all the text data that we are displaying on the UI with gettext function right?

Reply:
Right. But, it will work on direct strings not variables.